src/sources/prometheus/remote_write.rs

```diff
@@ -124,20 +124,11 @@ impl RemoteWriteSource {
 impl HttpSource for RemoteWriteSource {
     fn build_events(
         &self,
-        mut body: Bytes,
-        header_map: &HeaderMap,
+        body: Bytes,
+        _header_map: &HeaderMap,
         _query_parameters: &HashMap<String, String>,
         _full_path: &str,
     ) -> Result<Vec<Event>, ErrorMessage> {
-        // If `Content-Encoding` header isn't `snappy` HttpSource won't decode it for us
-        // se we need to.
-        if header_map
-            .get("Content-Encoding")
-            .map(|header| header.as_ref())
-            != Some(&b"snappy"[..])
-        {
-            body = decode(&Some("snappy".to_string()), body)?;
-        }
```

Comment:
This is a breaking change but I'm not sure if it should be here in the first place..

Reply:
This makes sense but I'm just afraid moving from a permissive state to a less permissive one may reveal client issues that are reported as Vector issue. I'd argue this could be a separate change from the others in this PR.

Reply:
Good find. For clarity, the reason I want to remove this code path is because it allows a client to send requests that count compressed bytes rather than uncompressed bytes, which results in miscounting bytes received.  Keeping this logic but counting uncompressed bytes would require further reworking.

Reply:
I updated the logic to default to snappy decoding in the correct place, resolving the issue mentioned above without introducing a breaking change.

Reply:
Yes that seems much cleaner.  A part of me feels that this looks like it was decoding as snappy, even if the header said gzip - but I recall looking into it and that wasn't the case. So I think we are all good..